hbase_sql/src/main/java/com/splicemachine/stream/RemoteQueryClientImpl.java

```diff
@@ -113,7 +114,12 @@ public void run() {
                         streamListener.completed(olapResult);
                     } catch (ExecutionException e) {
                         LOG.warn("Execution failed", e);
-                        streamListener.failed(e.getCause());
+                        Throwable cause = e.getCause();
+                        if (cause instanceof IOException || cause instanceof ConnectException) {
+                            streamListener.failed(StandardException.newException(SQLState.OLAP_SERVER_CONNECTION, cause));
```

Comment:
not sure if the exception is clear in the log, we could differentiate the log we print for these cases, e.g.

```
Throwable cause = e.getCause();
if (cause instanceof IOException || cause instanceof ConnectException) {
  LOG.warn("Execution failed: Lost OLAP Connection", e);
  streamListener.failed(StandardException.newException(SQLState.OLAP_SERVER_CONNECTION, cause));
} else {
  LOG.warn("Execution failed: Lost OLAP Connection", e);
  streamListener.failed(cause);
}
```

Reply:
In this case in the log you will see the wrapper exception with the code SQLState.OLAP_SERVER_CONNECTION, that is used fo such cases, and of course stack trace to navigate to the place